.github/workflows/lint-ts.yml

```diff
@@ -0,0 +1,31 @@
+name: Lint TS/JS
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
```

Comment:
let's also add a path to this workflow (following convention used in other lint workflows for this repo) to make sure the lint passes even if we do changes to this workflow